tables/atom.py

```diff
@@ -305,21 +305,33 @@ def from_sctype(cls, sctype, shape=(), dflt=None):
         sctype not represented in an Atom is ignored::
 
             >>> import numpy as np
-            >>> Atom.from_sctype(np.int16, shape=(2, 2))
-            Int16Atom(shape=(2, 2), dflt=0)
+            >>> (
+            ...  repr(Atom.from_sctype(np.int16, shape=(2, 2))) ==
+            ...  f"Int16Atom(shape=({np.int64(2)!r}, {np.int64(2)!r}), dflt={np.int16(0)!r})"
+            ... )
+            True
```

Comment:
As a general comment, I would prefer to keep the examples in the documentation as clean as possible.
If it is not possible to address it easily for both numpy1.x and 2.x then probably it is better to keep a clean docstring working for one of the numpy versions and maybe disable doctests for that function/module.

Reply:
Which one should I disable the doctests for?

Reply:
I'll disable the doctests on numpy 2